test/e2e/components/vl-select.js

```diff
@@ -35,47 +42,104 @@ class VlSelect extends VlElement {
             return this.findElements(By.css('option'));
         }
     }
-    
+
     async _mapDressedOptions(options) {
         return Promise.all(options.map(async (selectItem) => {
             const text = await selectItem.getText();
-            return {text: text, webElement: selectItem};
+            const webElement = await selectItem;
+            return { text: text, webElement: webElement };
+        }));
+    }
+
+    async _getWebelementMap() {
+        await this._openDressedDropdown();
+        const selectItems = await this._getOptions();
+        return this._mapDressedOptions(selectItems);
+    }
+
+    async _getValue(element) {
+        if ((await this.isDressed())) {
+            return element.getAttribute('data-value');
+        } else {
+            return element.getAttribute('value');
+        }
+    }
+
+    async _mapValues(options) {
+        return Promise.all(options.map(async (option) => {
+            const value = await this._getValue(option);
+            return { webElement: option, value: value }
+        }));
+    }
+
+    async _mapVisibleText(options) {
+        return Promise.all(options.map(async (option) => {
+            const textContent = await option.getAttribute('textContent');
+            const visibleText = textContent.replace(/\s+/g, ' ').trim();
+            return { webElement: option, visibleText: visibleText }
         }));
     }
-                            
+
+    async _clickOption(option) {
+        if ((await this.isDressed())) {
+            await this._openDressedDropdown();
+        }
+        return option.webElement.click();
+    }
+
     async isDressed() {
         return this.hasAttribute('data-vl-select-dressed');
     }
-    
+
     async values() {
         const options = await this._getOptions();
-        return Promise.all(options.map(o => o.getAttribute('textContent')));
+        return Promise.all(options.map(o => o.getAttribute('value')));
     }
 
     async hasValue(value) {
-        const values = (await this.values()).map(v => v.trim());
+        const values = await this.values();
         return values.includes(value);
     }
 
-    async selectValue(value) {
-        if(await !this.hasValue(value)) {
+    async getAllVisibleText() {
+        const options = await this._getOptions();
+        return Promise.all(options.map(o => o.getAttribute('textContent')));
+    }
+
+    async hasVisibleText(visibleText) {
+        const texts = (await this.getAllVisibleText()).map(t => t.trim());
+        return texts.includes(visibleText);
+    }
```

Comment:
visible is hier niet de juiste naam lijkt mij.

Reply:
Veranderd naar `getText` en `hasText`